src/main/java/net/fabricmc/loom/configuration/LoomDependencyManager.java

```diff
@@ -193,7 +194,9 @@ private static void handleInstallerJson(JsonObject jsonObject, Project project)
 
 			project.getLogger().debug("Loom adding " + name + " from installer JSON");
 
-			if (jsonElement.getAsJsonObject().has("url")) {
+			// If user choose to use dependencyResolutionManagement, then they should declare
+			// these repositories manually in the settings file.
+			if (jsonElement.getAsJsonObject().has("url") && !project.getGradle().getPlugins().hasPlugin(LoomRepositoryPlugin.class)) {
 				String url = jsonElement.getAsJsonObject().get("url").getAsString();
 				long count = project.getRepositories().stream().filter(artifactRepository -> artifactRepository instanceof MavenArtifactRepository)
```

Comment:
So here, when the count is 0, I think the best approach is to leave a warning or fail the build (maybe add a command-line option to control that) when the desired maven doesn't exist. Imo we should fail by default, but a system property for warning can allow people to use proxy mavens.

Reply:
The `settings.getDependencyResolutionManagement().getRepositories()` and `project.getRepositories()` are different object in gradle. Since we do not add any repo into `project.getRepositories()` if `dependencyResolutionManagement` used, the `project.getRepositories()` will always be empty. Furthermore, there is currently no way of retrieving `settings` in project lifecycle, this means we do not know if a repo is added or not. Thus, the `count` will always be zero. 

This limitation will cause your idea hard to implement, and hence I prefer the existing code (I really don't have any other solution for this).